File: orchalanguage-spring-integration-implementation/src/main/java/orcha/lang/compiler/referenceimpl/springIntegration/SpringIntegrationAutoConfiguration.java
package orcha.lang.compiler.referenceimpl.springIntegration;

import orcha.lang.compiler.WhenInstruction;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConditionalOnClass(WhenInstructionForSpringIntegration.class)
public class SpringIntegrationAutoConfiguration {

    @Bean(name = "whenInstruction")
    public WhenInstructionFactory whenInstructionFactory() {
        WhenInstructionFactory factory = new WhenInstructionFactory();
        return factory;
    }

    @Bean
    public WhenInstruction whenInstruction() throws Exception {
        return whenInstructionFactory().getObject();
    }
}

File: orchalang-spring-integration-implementation/src/main/kotlin/orcha/lang/compiler/referenceimpl/springIntegration/SpringIntegrationAutoConfiguration.kt
package orcha.lang.compiler.referenceimpl.springIntegration

import orcha.lang.compiler.Postprocessing
import orcha.lang.compiler.referenceimpl.PostprocessingImpl
import orcha.lang.compiler.syntax.SendInstruction
import orcha.lang.compiler.syntax.WhenInstruction
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
@ConditionalOnClass(WhenInstructionForSpringIntegration::class, SendInstructionForSpringIntegration::class)
class SpringIntegrationAutoConfiguration {

    @Bean(name = ["whenInstruction"])
    fun whenInstructionFactory(): WhenInstructionFactory {
        return WhenInstructionFactory()
    }

    @Bean
    @Throws(Exception::class)
    fun whenInstruction(): WhenInstruction? {
        return whenInstructionFactory().getObject()
    }

    @Bean(name = ["sendInstruction"])
    fun sendInstructionFactory(): SendInstructionFactory {
        return SendInstructionFactory()
    }

    @Bean
    @Throws(Exception::class)
    fun sendInstruction(): SendInstruction? {
        return sendInstructionFactory().getObject()
    }

    @Bean
    internal fun postprocessing(): Postprocessing {
        return PostprocessingImpl()
    }

}
